Revert "change implementation of Exiv2::base64encode() to adopt imple…

piponazo opened the pull request:
…mentation from same URL as base64decode()."

This reverts commit b596f311f4ee4c387421f1f3bab25b2107ce6b43.

Revert PR #1360

Commit:
Revert "change implementation of Exiv2::base64encode() to adopt implementation from same URL as base64decode()."

This reverts commit b596f311f4ee4c387421f1f3bab25b2107ce6b43.

clanmills commented:
The fix to compile base64_decode with Visual Studio is:

```diff
diff --git a/src/futils.cpp b/src/futils.cpp
index 82596d4b..a05e5021 100644
--- a/src/futils.cpp
+++ b/src/futils.cpp
@@ -189,7 +189,7 @@ namespace Exiv2 {
     }
 
     static std::string base64_decode(std::string const& encoded_string) {
-      int in_len = encoded_string.size();
+      int in_len = (int) encoded_string.size();
       int i = 0;
       int j = 0;
       int in_ = 0;
@@ -200,7 +200,7 @@ namespace Exiv2 {
         char_array_4[i++] = encoded_string[in_]; in_++;
         if (i ==4) {
           for (i = 0; i <4; i++)
-            char_array_4[i] = base64_chars.find(char_array_4[i]);
+            char_array_4[i] = (unsigned char) base64_chars.find(char_array_4[i]);
 
           char_array_3[0] = (char_array_4[0] << 2) + ((char_array_4[1] & 0x30) >> 4);
           char_array_3[1] = ((char_array_4[1] & 0xf) << 4) + ((char_array_4[2] & 0x3c) >> 2);
@@ -217,7 +217,7 @@ namespace Exiv2 {
           char_array_4[j] = 0;
 
         for (j = 0; j <4; j++)
-          char_array_4[j] = base64_chars.find(char_array_4[j]);
+          char_array_4[j] = (unsigned char) base64_chars.find(char_array_4[j]);
 
         char_array_3[0] = (char_array_4[0] << 2) + ((char_array_4[1] & 0x30) >> 4);
         char_array_3[1] = ((char_array_4[1] & 0xf) << 4) + ((char_array_4[2] & 0x3c) >> 2);
@@ -496,7 +496,7 @@ namespace Exiv2 {
         if ( procs    ) procstat_freeprocs(procstat, procs);
         if ( procstat ) procstat_close(procstat);
     #elif defined(__sun__)
-        // https://stackoverflow.com/questions/47472762/on-solaris-how-to-get-the-full-path-of-executable-of-running-process-programatic        
+        // https://stackoverflow.com/questions/47472762/on-solaris-how-to-get-the-full-path-of-executable-of-running-process-programatic
         const char* proc = Internal::stringFormat("/proc/%d/path/a.out",getpid()).c_str();
         char        path[500];
         ssize_t     l = readlink (proc,path,sizeof(path)-1);
```

I'm going to get out of the base64 code and leave @piponazo to manage this.  If/when you want something reviewed and approved, please let me know.

Thanks for working on this.  Another very busy week with Exiv2 comes to an end.

piponazo commented:
Appveyor just passed. The problem was that in 0.27-maintenance, somehow 2 commits modifying the file `futils.cpp` were merged ... and I just reverted 1 of them.

I just took the pervious version of `futils.cpp` before those commits and replaced the file here.

Once this PR is accepted and merged, I will re-open #1360 and add the tests.

clanmills reviewed (approved):
I'll approve this.  I may be quicker/simpler to not revert, but to patch base64encode() with the casts in the patch above.  Your choice.

I tried my patch and to my surprise the unit tests failed.  Ah, I didn't know about test_futils.cpp.

I'm going to keep out of the base64 code.  One cook is enough and @piponazo is a Master Chef!
